omod/src/main/java/org/openmrs/module/registrationapp/fragment/controller/RegisterPatientFragmentController.java

```diff
@@ -105,6 +107,7 @@ public FragmentActionResult importMpiPatient(@RequestParam("mpiPersonId") String
     public FragmentActionResult submit(UiSessionContext sessionContext, @RequestParam(value="appId") AppDescriptor app,
                             @SpringBean("registrationCoreService") RegistrationCoreService registrationService,
                             @ModelAttribute("patient") @BindParams Patient patient,
+                            @RequestParam(value = "patientIdNumber", required = false) String patientIdNumber,
```

Comment:
Why not try to inject the `Patient` instance directly like [here](https://github.com/openmrs/openmrs-module-coreapps/blob/8e38b095fb28d36ad397dd25ed6250b82c9331fc/omod/src/main/java/org/openmrs/module/coreapps/fragment/controller/EditPatientIdentifierFragmentController.java#L27)?

Reply:
Whatever you end up doing, make sure it works with patient **UUIDs**.